state_synchronizer/src/coordinator.rs

```diff
@@ -317,6 +320,11 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         peer_id: &PeerId,
         mut response: GetChunkResponse,
     ) -> Result<()> {
+        // we have received a response from the peer we requested a chunk, so
+        // reset the last_requested_peer_id field to avoid race condition on next peer
+        // to receive request
+        self.peer_manager.process_new_request(None);
```

Comment:
the function being called doesn't really feel to be aligned with the comment above. Maybe create two functions?
```rust
fn reset_last_requested_peer() { ... }
fn process_new_request(peer_id: PeerId) { ... }
```